Yet-Another-Pathfinding-Simulator/PriorityQueue.h

```diff
@@ -72,14 +72,16 @@ namespace yaps
             return Coordinates(-1, -1);
         }
 
-        T* removeHighestPrior()
+        T removeHighestPrior()
         {
-            if (NextFree == 0) return NULL;
-            Node zmienna = heap[0];
+            if (NextFree == 0) throw std::exception();
+            T returnVal = heap[0].item;
+
             heap[0] = heap[--NextFree];
-            heap.erase(heap.begin() + NextFree);
+            heap.erase(heap.begin());
```

Comment:
Why `+ NextFree` is removed?

Reply:
thats my mistake. Let me fix that up too.